data_safe_haven/resources/dns_server/dns_server_vm.cloud_init.mustache.yaml

```
@@ -0,0 +1,77 @@
+#cloud-config
+
+groups:
+  - docker
+
+write_files:
+  - encoding: b64
+    content: {{entrypoint_sh_content_encoded}}
+    path: "/opt/adguardhome/custom/entrypoint.sh"
+    permissions: "0644"
+
+  - encoding: b64
+    content: {{adguardhome_yaml_content_encoded}}
+    path: "/opt/adguardhome/custom/AdGuardHome.yaml"
+    permissions: "0644"
+
+  - path: "/etc/systemd/resolved.conf.d/adguardhome.conf"
+    permissions: "0644"
+    content: |
+      [Resolve]
+      DNS=127.0.0.1
+      DNSStubListener=no
+
+  - path: "/etc/systemd/system/adguardhome.service"
+    permissions: "0644"
+    content: |
+      [Unit]
+      Description=Starts the AdGuardHome container.
+      After=docker.service
+      Requires=docker.service
+
+      [Service]
+      TimeoutStartSec=0
+      Restart=always
+      Environment="HOME=/home/adguardhome"
+      ExecStartPre=/usr/bin/docker login -u {{dockerhub_username}} -p {{dockerhub_access_token}}
+      ExecStart=/usr/bin/docker run --rm --name adguardhome -v /opt/adguardhome/custom:/opt/adguardhome/custom -p 53:53/tcp -p 53:53/udp -p 80:80/tcp --entrypoint /bin/sh {{container_image}} /opt/adguardhome/custom/entrypoint.sh
```

```suggestion
      ExecStart=/usr/bin/docker run --rm --name adguardhome -v /opt/adguardhome/custom:/opt/adguardhome/custom -p 53:53/tcp -p 53:53/udp --entrypoint /bin/sh {{container_image}} /opt/adguardhome/custom/entrypoint.sh
```

I think this may have been because container instances required exposing port 80, we probably don't need it here.

---

I tried it, and the DNS server is working fine without exposing port 80. I'll make the change